pkg/controller/common/password/fixtures/fixtures.go

```diff
@@ -0,0 +1,42 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package fixtures
+
+import (
+	"context"
+
+	pwgenerator "github.com/m1/go-generate-password/generator"
+
+	commonpassword "github.com/elastic/cloud-on-k8s/v3/pkg/controller/common/password"
+)
+
+type testGenerator struct {
+	length int
+	commonpassword.RandomGenerator
+	RandomGeneratorWithSetter
+}
+
+type RandomGeneratorWithSetter interface {
+	commonpassword.RandomGenerator
+	SetLength(length int) commonpassword.RandomGenerator
+}
+
+func (t *testGenerator) Generate(ctx context.Context) ([]byte, error) {
+	generator, err := pwgenerator.NewWithDefault()
+	if err != nil {
+		return nil, err
+	}
+	data, err := generator.Generate()
+	return []byte(*data), err
+}
+
+func (t *testGenerator) SetLength(length int) commonpassword.RandomGenerator {
+	t.length = length
```

Comment:
Length is ignored during generation. 

How about just providing a test constructur for the real implementation instead?
```golang

func MustTestRandomGenerator(length int) commonpassword.RandomGenerator {
	generator, err := commonpassword.NewRandomPasswordGenerator(
		commonpassword.GeneratorParams{
			Length: length,
		},
		func(ctx context.Context) (bool, error) {
			return true, nil
		},
	)
	if err != nil {
		panic(err) // acceptable for test fixtures
	}
	return generator
}
```